client/question/domain/CodeEvalResultObjectFactory.js

```diff
@@ -143,8 +159,7 @@ tie.factory('CodeEvalResultObjectFactory', [
      * @returns {boolean}
      */
     CodeEvalResult.prototype.hasTimeLimitError = function() {
-      var errorString = this.getErrorString();
-      return (errorString && errorString.startsWith('TimeLimitError'));
+      return this.getErrorString() && this._timeLimitExceeded;
```

Comment:
Do you actually need the this.getErrorString() bit?

Reply:
Nope!